src/controllers/bookController.js

```diff
@@ -0,0 +1,195 @@
+    }
+
+    const books = await Book.find(queryObj)
+      .sort(sort)
+      .limit(Number(limit))
+      .skip(Number(skip));
+
+    res.status(StatusCodes.OK).json({ books, count: books.length });
+  } catch (error) {
+    console.error('Error fetching books:', error);
+    next(error);
+  }
+};
+
+const getBook = async (req, res, next) => {
+  try {
+    const { id: bookId } = req.params;
+
+    const book = await Book.findById(bookId);
+
+    if (!book) {
+      return next(new NotFoundError(`No book with id ${bookId}`));
+    }
+
+    res.status(StatusCodes.OK).json({ book });
+  } catch (error) {
+    console.error('Error fetching book:', error);
+    next(error);
+  }
+};
+
+const createBook = async (req, res, next) => {
+  try {
+    const { coverImageUrl, ...bookData } = req.body;
+    bookData.createdBy = req.user.userId;
+    bookData.language = req.body.language || 'English';
+    bookData.coverImageUrl =
+      coverImageUrl || process.env.DEFAULT_COVER_IMAGE_URL;
+
+    const book = await Book.create(bookData);
+    res
+      .status(StatusCodes.CREATED)
+      .json({ msg: 'The book has been successfully created.', book });
+  } catch (error) {
+    next(error);
+  }
+};
+
+const updateBook = async (req, res, next) => {
+  try {
+    const { coverImageUrl, ...updateData } = req.body;
+    const { id: bookId } = req.params;
+    const {
+      user: { userId },
+    } = req;
+    
+    const book = await Book.findById(bookId);
+    if (!book) {
+      return next(new NotFoundError(`No book found with id: ${bookId}`));
+    }
+
+    if (book.createdBy.toString() !== userId) {
+      return next(
+        new BadRequestError('You are not authorized to update this book')
+      );
+    }
+
+    if (
+      coverImageUrl &&
+      coverImageUrl !== process.env.DEFAULT_COVER_IMAGE_URL
+    ) {
+      updateData.coverImageUrl = coverImageUrl;
+    } else if (!coverImageUrl) {
+      updateData.coverImageUrl = book.coverImageUrl;
+    }
+
+    const updatedBook = await Book.findByIdAndUpdate(bookId, updateData, {
+      new: true,
+      runValidators: true,
+    });
+
+    if (!updatedBook) {
+      console.log('Failed to update book');
+      return next(new Error('Book update failed.'));
+    }
+
+    res.status(StatusCodes.OK).json({ msg: 'The book has been successfully updated.', updatedBook });
+  } catch (error) {
+    console.error('Error updating book:', error);
+    next(error);
+  }
+};
+
+const deleteBook = async (req, res, next) => {
+  try {
+    const {
+      user: { userId },
+      params: { id: bookId },
+    } = req;
+
+    const book = await Book.findById(bookId);
+    if (!book) {
+      return next(new NotFoundError(`No book found with id: ${bookId}`));
+    }
+
+    if (book.createdBy.toString() !== userId) {
+      return next(new BadRequestError('You are not authorized to delete this book'));
+    }
+
+    await Book.findOneAndDelete({ _id: bookId, createdBy: userId });
```

Comment:
We don't need created by here, since the bookId will be uique and is fectched above(const book = await Book.findById(bookId); you can use the book._id 

*but* we should just use findbyIDandDelete and only make one API call.